os/arch/arm/src/common/up_checkstack.c

```diff
@@ -72,6 +72,25 @@
 
 #ifdef CONFIG_STACK_COLORATION
 
+#ifndef CONFIG_STACK_ALIGNMENT
+
+/* The symbol  __ARM_EABI__ is defined by GCC if EABI is being used.  If you
+ * are not using GCC, make sure that CONFIG_STACK_ALIGNMENT is set correctly!
+ */
+
+#ifdef __ARM_EABI__
+#define CONFIG_STACK_ALIGNMENT 8
+#else
+#define CONFIG_STACK_ALIGNMENT 4
+#endif
+#endif
+
+/* Stack alignment macros */
+
+#define STACK_ALIGN_MASK    (CONFIG_STACK_ALIGNMENT - 1)
+#define STACK_ALIGN_DOWN(a) ((a) & ~STACK_ALIGN_MASK)
+#define STACK_ALIGN_UP(a)   (((a) + STACK_ALIGN_MASK) & ~STACK_ALIGN_MASK)
+
```

Comment:
I think it's better to define them in a common header instead of duplicating the code. 

https://github.com/Samsung/TizenRT/blob/29151cd53f5fe15e1a788689d1a881435f0ad5ae/os/arch/arm/src/common/up_createstack.c#L92-L109